core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/datalake/upgrade/validation/ClusterUpgradeValidationActions.java

```diff
@@ -0,0 +1,135 @@
+                        payload.getResourceId(), payload.getImageId());
+                sendEvent(context, event.selector(), event);
+            }
+
+        };
+    }
+
+    @Bean(name = "CLUSTER_UPGRADE_DISK_SPACE_VALIDATION_STATE")
+    public Action<?, ?> clusterUpgradeDiskSpaceValidation() {
+        return new AbstractClusterUpgradeValidationActions<>(ClusterUpgradeValidationEvent.class) {
+
+            @Override
+            protected void doExecute(CommonContext context, ClusterUpgradeValidationEvent payload, Map<Object, Object> variables) {
+                LOGGER.info("Starting disk space validation.");
+                ClusterUpgradeValidationEvent event = new ClusterUpgradeValidationEvent(VALIDATE_CLUSTER_EVENT.name(),
+                        payload.getResourceId(), payload.getImageId());
+                sendEvent(context, event.selector(), event);
+            }
+
+        };
+    }
+
+    @Bean(name = "CLUSTER_UPGRADE_VALIDATION_FINISHED_STATE")
+    public Action<?, ?> clusterUpgradeValidationFinished() {
+        return new AbstractClusterUpgradeValidationActions<>(ClusterUpgradeValidationEvent.class) {
+
+            @Override
+            protected void doExecute(CommonContext context, ClusterUpgradeValidationEvent payload, Map<Object, Object> variables) {
+                LOGGER.info("Cluster upgrade validation finished.");
+                Long resourceId = payload.getResourceId();
+                cloudbreakEventService.fireCloudbreakEvent(resourceId, AVAILABLE.name(), ResourceEvent.CLUSTER_UPGRADE_VALIDATION_FINISHED);
+                ClusterUpgradeValidationEvent event = new ClusterUpgradeValidationEvent(FINALIZE_CLUSTER_UPGRADE_VALIDATION_EVENT.name(),
+                        payload.getResourceId(), payload.getImageId());
+                sendEvent(context, event);
+            }
+
+        };
+    }
+
+    @Bean(name = "CLUSTER_UPGRADE_VALIDATION_FAILED_STATE")
+    public Action<?, ?> clusterUpgradeValidationFailed() {
+        return new AbstractClusterUpgradeValidationActions<>(ClusterUpgradeValidationFailureEvent.class) {
+
+            @Override
+            protected void doExecute(CommonContext context, ClusterUpgradeValidationFailureEvent payload, Map<Object, Object> variables) {
+                String errorMessage = payload.getException().getMessage();
+                Long resourceId = payload.getResourceId();
+                if (payload.getException() instanceof UpgradeValidationFailedException) {
+                    LOGGER.error("Cluster upgrade validation failed with validation error: {}", errorMessage);
+                    cloudbreakEventService.fireCloudbreakEvent(resourceId, AVAILABLE.name(), ResourceEvent.CLUSTER_UPGRADE_VALIDATION_FAILED,
+                            Collections.singletonList(errorMessage));
+                    sendEvent(context);
```

Comment:
This causes an endless loop for me locally. This step is missing the `createRequest` method.